src/actors/cficaches.rs

```diff
@@ -104,46 +112,52 @@ impl CacheItemRequest for FetchCfiCacheInternal {
     type Error = CfiCacheError;
 
     fn get_cache_key(&self) -> CacheKey {
-        self.object_meta.cache_key()
+        self.meta_handle.cache_key()
     }
 
     /// Extracts the Call Frame Information (CFI) from an object file.
     ///
     /// The extracted CFI is written to `path` in symbolic's
     /// [`CfiCache`](symbolic::minidump::cfi::CfiCache) format.
     fn compute(&self, path: &Path) -> BoxedFuture<Result<CacheStatus, Self::Error>> {
-        let path = path.to_owned();
-        let object = self
-            .objects_actor
-            .fetch(self.object_meta.clone())
-            .map_err(CfiCacheError::Fetching);
-
+        let objects_actor = self.objects_actor.clone();
+        let meta_handle = self.meta_handle.clone();
         let threadpool = self.threadpool.clone();
-        let result = object.and_then(move |object| {
-            let future = future::lazy(move |_| {
-                if object.status() != CacheStatus::Positive {
-                    return Ok(object.status());
-                }
-
-                let status = if let Err(e) = write_cficache(&path, &*object) {
-                    log::warn!("Could not write cficache: {}", e);
-                    sentry::capture_error(&e);
+        let path = path.to_owned();
 
-                    CacheStatus::Malformed
-                } else {
-                    CacheStatus::Positive
+        let result = async move {
+            let object_handle = objects_actor
+                .fetch(meta_handle.clone())
+                .await
+                .map_err(CfiCacheError::Fetching)?;
+
+            let future = async move {
+                let status = match object_handle.status() {
+                    CacheStatus::Positive => match write_cficache(&path, &*object_handle) {
+                        Ok(()) => CacheStatus::Positive,
+                        Err(err) => {
+                            log::warn!("Could not write cficache: {}", err);
+                            sentry::capture_error(&err);
+                            CacheStatus::Malformed
+                        }
+                    },
+                    cache_status => cache_status,
```

Comment:
I preferred the former structure as it was easier to read and had less indentation.